### PR TITLE
[RTIR] Support Token and Certificate authentication

### PR DESCRIPTION
**Pull request by mmhw**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-8721)

## Description
- Replaced the display of **Credentials** parameter to **Username**.
- Added the **Token** parameter to use API Token authentication.
- Added the **Certificate** and **Private Key** parameters to use Certificate authentication.

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/RTIR/Integrations/RTIR/RTIR.yml [hunk 1]
```diff
@@ -58,336 +67,336 @@ script:
...
+    - description: The ticket content.
       name: text
-    - description: Ticket initial priority, given as number in the range 0-100
+    - description: Ticket initial priority, given as number in the range 0-100.
       name: initial-priority
-    - description: Ticket final priority, given as number in the range 0-100
+    - description: Ticket final priority, given as number in the range 0-100.
       name: final-priority
-    - description: Attachment to add to the ticket, given as Entry ID. Comma separated values supported, e.g. entryID1,entryID2
+    - description: Attachment to add to the ticket, given as Entry ID. Comma separated values supported, e.g. entryID1,entryID2.
```
Comment (ShirleyDenkberg) [inline Packs/RTIR/Integrations/RTIR/RTIR.yml:93]:
```suggestion
    - description: Attachment to add to the ticket, given as Entry ID. Comma-separated values supported, e.g., entryID1,entryID2.
```

File: Packs/RTIR/Integrations/RTIR/RTIR.yml [hunk 2]
```diff
@@ -58,336 +67,336 @@ script:
...
       type: string
   - arguments:
-    - description: 'Ticket ID '
+    - description: 'Ticket ID.'
       name: ticket-id
-    - description: Ticket subject
+    - description: Ticket subject.
       name: subject
-    - description: Ticket queue (Comma separated values supported, e.g. General,Incident reports,Incidents)
+    - description: Ticket queue (Comma separated values supported, e.g. General,Incident reports,Incidents).
```
Comment (ShirleyDenkberg) [inline Packs/RTIR/Integrations/RTIR/RTIR.yml:127]:
```suggestion
    - description: Ticket queue (Comma-separated values supported, e.g., General,Incident reports,Incidents).
```

File: Packs/RTIR/Integrations/RTIR/RTIR.yml [hunk 3]
```diff
@@ -58,336 +67,336 @@ script:
...
-    - description: Ticket ID to add comment to
+    - description: Ticket ID to add comment to.
       name: ticket-id
       required: true
-    - description: Comment text
+    - description: Comment text.
       name: text
       required: true
-    - description: Attachment to add to the ticket, given as Entry ID. Comma separated values supported, e.g. entryID1,entryID2
+    - description: Attachment to add to the ticket, given as Entry ID. Comma separated values supported, e.g. entryID1,entryID2.
```
Comment (ShirleyDenkberg) [inline Packs/RTIR/Integrations/RTIR/RTIR.yml:378]:
```suggestion
    - description: Attachment to add to the ticket, given as Entry ID. Comma-separated values supported, e.g., entryID1,entryID2.
```

File: Packs/RTIR/Integrations/RTIR/RTIR_description.md [hunk 4]
```diff
@@ -1,3 +1,16 @@
-Additional Parameters:
+**Authentication Methods**
+- *Basic Auth* - Authenticates using a Username and Password.
+- *Token Auth* - Authenticates using a Token instead of Username and Password.
+
+**Certificate**
+If using certificate authentication, add the Certificate and Private Key parameters.
+To get the certificate and private key from Pkcs12 (.p12) file:
```
Comment (ShirleyDenkberg) [inline Packs/RTIR/Integrations/RTIR/RTIR_description.md:7]:
```suggestion
To get the certificate and private key from the PKCS12 (.p12) file:
```

File: Packs/RTIR/Integrations/RTIR/RTIR_description.md [hunk 5]
```diff
@@ -1,3 +1,16 @@
...
+- Run on your terminal `openssl pkcs12 -clcerts -nokeys -in [your_certificate_name.p12] -out [certificate_name.txt]` to extract certificate from .p12 file.
```
Comment (ShirleyDenkberg) [inline Packs/RTIR/Integrations/RTIR/RTIR_description.md:8]:
```suggestion
- On your terminal, run `openssl pkcs12 -clcerts -nokeys -in [your_certificate_name.p12] -out [certificate_name.txt]` to extract the certificate from the .p12 file.
```

File: Packs/RTIR/Integrations/RTIR/RTIR_description.md [hunk 6]
```diff
@@ -1,3 +1,16 @@
...
+- Open the [certificate_name.txt] file and copy the contents (only the certificate part) to the `Certificate` parameter.
+- Delete the [certificate_name.txt] file after copying the contents.
+- Run on your terminal `openssl pkcs12 -in [your_certificate_name.p12] -nodes -nocerts -out [private_key_name.txt]` to extract private key from .p12 file.
```
Comment (ShirleyDenkberg) [inline Packs/RTIR/Integrations/RTIR/RTIR_description.md:11]:
```suggestion
- On your terminal, run `openssl pkcs12 -in [your_certificate_name.p12] -nodes -nocerts -out [private_key_name.txt]` to extract the private key from the .p12 file.
```

File: Packs/RTIR/Integrations/RTIR/RTIR_description.md [hunk 7]
```diff
@@ -1,3 +1,16 @@
...
+- Open the [private_key_name.txt] file and copy the contents (only the private key part) to the `Private Key` parameter.
+- Deleted the [private_key_name.txt] file after copying the contents.
```
Comment (ShirleyDenkberg) [inline Packs/RTIR/Integrations/RTIR/RTIR_description.md:13]:
```suggestion
- Delete the [private_key_name.txt] file after copying the contents.
```

File: Packs/RTIR/ReleaseNotes/1_0_17.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+
+#### Integrations
+
+##### RTIR
+
+- Replaced the display of **Credentials** parameter to **Username**.
+- Added the **Token** parameter to use API Token authentication.
+- Added the **Certificate** and **Private Key** parameters to use Certificate authentication.
```
Comment (ShirleyDenkberg) [inline Packs/RTIR/ReleaseNotes/1_0_17.md:8]:
```suggestion
- Replaced the display of *Credentials* parameter to *Username*.
- Added the *Token* parameter to use API Token authentication.
- Added the *Certificate* and *Private Key* parameters to use Certificate authentication.
```

**Comment by ShirleyDenkberg** [comment]
@DeanArbel Doc review completed.

**Comment by mmhw** [comment]
> @DeanArbel Doc review completed.

Thanks!

**Review by DeanArbel — APPROVED** [review]
Very nice job, good work.
Please make sure to work with the tech docs team, as there are many changes introduced.

**Pull request merged**